spec/repository_spec.rb

```diff
@@ -35,37 +41,71 @@
         end.converge(described_recipe)
       end
 
-      it 'sets the yumrepo_gpgkey_new attribute' do
-        expect(chef_run.node['datadog']['yumrepo_gpgkey_new']).to match(
+      # Key E09422B3
+      it 'sets the yumrepo_gpgkey_new attribute E09422B3' do
+        expect(chef_run.node['datadog']['yumrepo_gpgkey_new_e09422b3']).to match(
           /DATADOG_RPM_KEY_E09422B3.public/
         )
       end
 
-      it 'installs gnupg' do
+      it 'installs gnupg E09422B3' do
         expect(chef_run).to install_package('gnupg') if chef_run.node['packages']['gnupg2'].nil?
       end
 
-      it 'downloads the new RPM key' do
+      it 'downloads the new RPM key E09422B3' do
         expect(chef_run).to create_remote_file('DATADOG_RPM_KEY_E09422B3.public').with(path: ::File.join(Chef::Config[:file_cache_path], 'DATADOG_RPM_KEY_E09422B3.public'))
       end
 
-      it 'notifies the GPG key install if a new one is downloaded' do
+      it 'notifies the GPG key install if a new one is downloaded E09422B3' do
         keyfile_r = chef_run.remote_file('DATADOG_RPM_KEY_E09422B3.public')
         expect(keyfile_r).to notify('execute[rpm-import datadog key e09422b3]')
           .to(:run).immediately
       end
 
-      it 'doesn\'t execute[rpm-import datadog key e09422b3] by default' do
+      it 'doesn\'t execute[rpm-import datadog key *] by default E09422B3' do
         keyfile_exec_r = chef_run.execute('rpm-import datadog key e09422b3')
         expect(keyfile_exec_r).to do_nothing
       end
 
       # prefer HTTPS on boxes that support TLS1.2
-      it 'sets up a yum repo' do
+      it 'sets up a yum repo E09422B3' do
         expect(chef_run).to create_yum_repository('datadog').with(
           gpgkey: 'https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public'
         )
       end
+
+      # Key 20200908
+      it 'sets the yumrepo_gpgkey_new attribute 20200908' do
```

Comment:
I don't have a strong opinion on whether the test names should have the key date or the key fingerprint. Keeping the date is fine to me.